docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,596 @@
+
+3. Delete line 17 (TODO: verify line number):
+    ```python
+    items_names = list(grocery_list.values())
+    ```
+    And replace it with:
+    ```python
+      item_id = redis_client.hget("item_name_to_id", item.item_name)
+     ```
+
+Note that Pylance raises a problem with this change. This is because the `hget` method returns either `str`, or `None` (if the item doesn’t exist). However, the lines below our code that we haven’t replaced yet expect `item_id` to be of type `int`. Let’s address it by renaming the `item_id` symbol.
+
+4. Rename `item_id` with `item_id_str` by selecting the symbol and running the **Rename Symbol** command (`kb(editor.action.rename)`).
+
+    This will rename all occurrences of the symbol in the file, including the one in the `return` statement.
+
+5. Pylance should show a variable type hint next to `item_id_str`. Double click to accept it:
+    ![Variable type hint displayed next to the item_id_str variable.](images/fastapi-tutorial/pylance_redis_typehint.png)
+
+6. If the item doesn't exist, then `item_id_str` will be None. So now we can delete line 18 (TODO: verify line):
+    ```python
+    if item.item_name in items_names:
+    ```
+    And replace it with:
+    ```python
+    if item_id_str is not None:
+    ```
+Now that we have the `item_id` in a string, we just need to convert it to an `int` and add the quantity provided to the item. Because the redis hash we have so far only maps item names to their IDs, we now need to to map item IDs to their names and quantity. One way to do that is to create hashes for each item, in the format `"item_id:{item_id}"`, and provide it with "name" and "quantity" fields.
+
+7.	Replace lines 19 and 20 with the following, to convert the `item_id` to an `int`, and then incrementing the quantity of our item by calling the `hincrby` method from redis:
+    ```python
+    item_id = int(item_id_bytes.decode())
+    redis_client.hincrby(f"item_id:{item_id}", "quantity", item.quantity)
+    ```
+
+We now only need to replace the code for when the item does not exist, i.e. `item_id_str` is None. In this case, we will generate a new `item_id`, create a new redis hash for our item, and then add the provided item names and quantity.
+To generate a new `item_id`, let’s use the `incr` method from redis, passing a new hash "item_ids". When this method is run for the first time, it will create the item_ids hash with a unique number, and then each time it's run it will generate an incremental number and store it in this hash.
+
+8. Delete line 21 (TODO: verify line):
+    ```python
+    item_id: int = len(grocery_list)
+    ```
+    And add the following:
+    ```python
+    item_id: int = redis_client.incr("item_ids")
+    ```
+
+Now we will add our item to our redis hash, using the `hset` method and by providing a mapping of fields or keys (`item_id`, `quantity` and `name`), and the values (the item’s newly created id, and its provided name and quantity).
```

Comment:
Here you use "id" but on line 407 you use "IDs" (capitalized). I personally prefer lowercase, but maybe the vscode docs have a standard?

Reply:
Great catch. It seems in our other docs we use IDs so I went with that